Automattic/jetpack
# My Jetpack: Add license key selector in License Activation screen.

Fixes #22913

Currently, the Activation UI shows an input box where users have to manually type a license key to activate a product, as shown in the attached capture.

We would like to replace the input box with a dropdown that list the user's available license keys and supports manually entering a key also.

**Before**
<img width="1109" alt="Screen Shot 2022-11-25 at 9 41 23 PM" src="https://user-images.githubusercontent.com/56598660/203997814-e28ea332-0fe3-4df4-b7bb-83ae0c0d6570.png">

**After**
<img width="1117" alt="Screen Shot 2022-11-25 at 9 42 13 PM" src="https://user-images.githubusercontent.com/56598660/203998009-bee013a4-0b73-465c-bf72-39cb6e786d64.png">


#### Changes proposed in this Pull Request:

* Replace the input box on the Activation UI with a dropdown that shows the user's available licenses and supports entering a license key manually

#### Other information:


#### Jetpack product discussion
Context: 1202858161751496-as-1201858908716176

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
1. Launch a test site that uses a build of the Jetpack plugin based on this branch
2. Connect your site to Jetpack, if it's not already
3. Make sure you do not have a jetpack license key that is not yet activated
4. Go to My Jetpack > Scroll down and click Activate a license
<img width="399" alt="Screen Shot 2022-11-25 at 9 48 39 PM" src="https://user-images.githubusercontent.com/56598660/203999156-74d19d0e-9d32-479b-a95e-ba235dd811a3.png">
5. Confirm that there is no license key selector rendered (only the input box).
<img width="1109" alt="Screen Shot 2022-11-25 at 9 41 23 PM" src="https://user-images.githubusercontent.com/56598660/203999821-1cb7f25d-324a-4036-aee8-518f5d919f11.png">

6. Open another tab go to https://cloud.jetpack.com/pricing and purchase any Jetpack product (make sure you do not activate it to any site)
7. Go back to the license key activation screen and refresh the page
8. confirm that the screen renders a license key selection.
<img width="1117" alt="Screen Shot 2022-11-25 at 9 42 13 PM" src="https://user-images.githubusercontent.com/56598660/203999787-9bd10f41-6436-460b-b453-7d1e3a284ea4.png">
9. select the new license key you purchase and activate. Confirm that the license key **successfully activated**.


#### Testing manual input
1. Purchase another product in the pricing page https://cloud.jetpack.com/pricing
2. Go back to the license key activation screen and select 'I want to add a license key manually'.
<img width="1008" alt="204000743-94cc9754-6a4b-4f6e-9b03-1dd74751c28d" src="https://user-images.githubusercontent.com/56598660/204257116-334d4c7a-ec41-499f-94d1-5930e3665473.png">

3. Confirm that the license key **input box** is rendered.
<img width="1105" alt="Screen Shot 2022-12-06 at 2 37 52 PM" src="https://user-images.githubusercontent.com/56598660/205839267-d405d38b-5484-48ce-81ab-b1465b61a9dc.png">





---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1201858908716176
